File: ElectroRemote/UI/Screens/ShopScreen.swift
import SwiftUI

/// Магазин аксессуаров и доп. оборудования: витрина карточками, карточка товара с
/// описанием и заявкой (телефон + комментарий). Оплаты в приложении нет — менеджер
/// перезванивает. Без входа витрину видно, заказать — только после входа.
struct ShopScreen: View {
    @Environment(\.palette) private var p
    let products: [ProductDto]
    let loggedIn: Bool
    let model: String?
    let onOrder: (_ productId: String, _ qty: Int, _ phone: String, _ comment: String, _ done: @escaping (String?) -> Void) -> Void
    let onBack: () -> Void
    let onRefresh: () -> Void

    @State private var category = "all"
    @State private var selected: ProductDto? = nil

    private var categories: [(String, String)] {
        [("all", L("Все")), ("accessory", L("Аксессуары")), ("equipment", L("Оборудование")),
         ("care", L("Уход")), ("tuning", L("Тюнинг")), ("other", L("Другое"))]
    }

    var body: some View {
        // сначала товары для своей модели, потом общие, потом остальные
        let m = (model ?? "").uppercased()
        let shown = products
            .filter { category == "all" || $0.category == category }
            .sorted { a, b in rank(a, m) < rank(b, m) }
        let columns = [GridItem(.flexible(), spacing: Space.x3), GridItem(.flexible(), spacing: Space.x3)]
        ScreenScaffold(title: L("Магазин"), onBack: onBack) {
            ScrollView(.horizontal, showsIndicators: false) {
                HStack(spacing: 6) {
                    ForEach(categories, id: \.0) { c in
                        let on = category == c.0
                        Button { category = c.0 } label: {
                            Text(c.1).font(ElectroType.body).foregroundStyle(on ? p.accent : p.textPrimary)
                                .padding(.horizontal, Space.x4).frame(height: ControlSize.chip)
                                .background(on ? p.accent.opacity(0.14) : p.surfaceElevated)
                                .clipShape(Capsule())
                                .overlay(Capsule().stroke(on ? p.accent : .clear, lineWidth: 1))
                        }
                        .buttonStyle(.plain)
                    }
                }
            }
            if shown.isEmpty {
                EmptyNote(text: products.isEmpty ? L("Товары скоро появятся.") : L("В этом разделе пусто."))
            }
            LazyVGrid(columns: columns, spacing: Space.x3) {
                ForEach(shown) { pr in
                    ProductCard(product: pr) { selected = pr }
                }
            }
        }
        .refreshable { onRefresh() }
        .onAppear(perform: onRefresh)
        .fullScreenCover(item: $selected) { pr in
            ProductScreen(product: pr, loggedIn: loggedIn, onOrder: onOrder) { selected = nil }
        }
    }

    private func rank(_ pr: ProductDto, _ m: String) -> Int {
        if pr.models.isEmpty { return 1 }
        if !m.isEmpty && pr.models.uppercased().contains(m) { return 0 }
        return 2
    }
}

func formatPrice(_ price: Int64, _ currency: String) -> String {
    let f = NumberFormatter(); f.numberStyle = .decimal; f.groupingSeparator = " "; f.usesGroupingSeparator = true
    let n = f.string(from: NSNumber(value: price)) ?? "\(price)"
    return currency == "USD" ? "$\(n)" : L("{0} сум", n)
}

private struct ProductCard: View {
    @Environment(\.palette) private var p
    let product: ProductDto
    let onTap: () -> Void

    var body: some View {
        let discount = (product.old_price ?? 0) > product.price
        Button(action: onTap) {
            VStack(alignment: .leading, spacing: 4) {
                ZStack(alignment: .topLeading) {
                    Group {
                        if let s = product.image_url, let u = URL(string: s) {
                            AsyncImage(url: u) { phase in
                                if let img = phase.image { img.resizable().scaledToFill() } else { p.surfaceElevated }
                            }
                        } else {
                            ZStack { p.surfaceElevated; Image(systemName: "bag").font(.system(size: 30)).foregroundStyle(p.textMuted) }
                        }
                    }
                    .aspectRatio(1, contentMode: .fill).frame(maxWidth: .infinity).clipped()
                    .clipShape(RoundedRectangle(cornerRadius: Radius.sm, style: .continuous))
                    if discount, let old = product.old_price {
                        Text("-\(100 - Int(product.price * 100 / old))%")
                            .font(.system(size: 10, weight: .bold)).foregroundStyle(p.onAccent)
                            .padding(.horizontal, 7).padding(.vertical, 2).background(p.accent).clipShape(Capsule())
                            .padding(8)
                    }
                }
                Text(product.title).font(ElectroType.body.weight(.semibold)).foregroundStyle(p.textPrimary)
                    .lineLimit(2).frame(minHeight: 40, alignment: .topLeading).padding(.top, 4)
                Text(formatPrice(product.price, product.currency)).font(ElectroType.body.weight(.bold)).foregroundStyle(p.accent)
                if discount, let old = product.old_price {
                    Text(formatPrice(old, product.currency)).font(ElectroType.unit).foregroundStyle(p.textMuted).strikethrough()
                }
            }
            .padding(Space.x3)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(p.surface)
            .clipShape(RoundedRectangle(cornerRadius: Radius.md, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: Radius.md, style: .continuous).stroke(p.outline, lineWidth: 1))
        }
        .buttonStyle(.plain)
    }
}

/// Карточка товара: фото, цена, описание, заявка.
private struct ProductScreen: View {
    @Environment(\.palette) private var p
    @Environment(\.openURL) private var openURL
    let product: ProductDto
    let loggedIn: Bool
    let onOrder: (String, Int, String, String, @escaping (String?) -> Void) -> Void
    let onClose: () -> Void

    @State private var qty = 1
    @State private var phone = ""
    @State private var comment = ""
    @State private var busy = false
    @State private var sent = false
    @State private var error: String? = nil
    @State private var showForm = false

    var body: some View {
        let discount = (product.old_price ?? 0) > product.price
        VStack(spacing: 0) {
            HStack(spacing: Space.x3) {
                Button(action: onClose) {
                    Image(systemName: "chevron.left").font(.system(size: 20, weight: .medium)).foregroundStyle(p.textPrimary).frame(width: 32, height: 32)
                }
                .buttonStyle(.plain)
                Text(L("Товар")).font(ElectroType.headline).foregroundStyle(p.textPrimary)
                Spacer()
            }
            .padding(Space.x4)
            ScrollView {
                VStack(alignment: .leading, spacing: Space.x4) {
                    if let s = product.image_url, let u = URL(string: s) {
                        AsyncImage(url: u) { phase in
                            if let img = phase.image { img.resizable().scaledToFill() } else { p.surfaceElevated }
                        }
                        .frame(maxWidth: .infinity).frame(height: 240).clipped()
                        .clipShape(RoundedRectangle(cornerRadius: Radius.md, style: .continuous))
                    }
                    Text(product.title).font(ElectroType.title).foregroundStyle(p.textPrimary)
                    HStack(alignment: .lastTextBaseline, spacing: Space.x2) {
                        Text(formatPrice(product.price, product.currency)).font(ElectroType.headline).foregroundStyle(p.accent)
                        if discount, let old = product.old_price {
                            Text(formatPrice(old, product.currency)).font(ElectroType.caption).foregroundStyle(p.textMuted).strikethrough()
                        }
                    }
                    if !product.models.isEmpty {
                        Text(L("Подходит: {0}", product.models)).font(ElectroType.caption).foregroundStyle(p.textSecondary)
                    }
                    if !product.description.isEmpty {
                        Text(product.description).font(ElectroType.body).foregroundStyle(p.textSecondary).fixedSize(horizontal: false, vertical: true)
                    }
                    if let l = product.link, let u = URL(string: l) {
                        Button { openURL(u) } label: {
                            Text(L("Подробнее на сайте")).font(ElectroType.body.weight(.semibold)).foregroundStyle(p.accent)
                        }
                        .buttonStyle(.plain)
                    }
                    if sent {
                        ElectroToast(kind: .success, title: L("Заявка отправлена"), message: L("Менеджер свяжется с вами по указанному телефону."))
                    } else if showForm {
                        VStack(alignment: .leading, spacing: Space.x3) {
                            Text(L("Заявка на покупку")).font(ElectroType.body.weight(.semibold)).foregroundStyle(p.textPrimary)
                            HStack {
                                Text(L("Количество")).font(ElectroType.body).foregroundStyle(p.textSecondary)
                                Spacer()
                                qtyButton("minus") { if qty > 1 { qty -= 1 } }
                                Text("\(qty)").font(ElectroType.headline).foregroundStyle(p.textPrimary).padding(.horizontal, Space.x3)
                                qtyButton("plus") { if qty < 99 { qty += 1 } }
                            }
                            AuthField(label: L("ТЕЛЕФОН"), value: $phone, keyboard: .phonePad, contentType: .telephoneNumber)
                            AuthField(label: L("КОММЕНТАРИЙ (необязательно)"), value: $comment, autocapitalize: true)
                            if let e = error { Text(e).font(ElectroType.caption).foregroundStyle(p.danger) }
                            ElectroButton(text: busy ? L("Отправляем…") : L("Отправить заявку"), enabled: !busy && phone.trimmingCharacters(in: .whitespaces).count >= 5) {
                                busy = true; error = nil
                                onOrder(product.id, qty, phone.trimmingCharacters(in: .whitespaces), comment.trimmingCharacters(in: .whitespaces)) { err in
                                    busy = false
                                    if let err { error = err } else { sent = true }
                                }
                            }
                        }
                        .padding(Space.x4)
                        .background(p.surface)
                        .clipShape(RoundedRectangle(cornerRadius: Radius.md, style: .continuous))
                        .overlay(RoundedRectangle(cornerRadius: Radius.md, style: .continuous).stroke(p.outline, lineWidth: 1))
                    }
                }
                .padding(.horizontal, Space.x5)
                .padding(.bottom, Space.x6)
            }
            .scrollDismissesKeyboard(.interactively)
            if !sent && !showForm {
                if loggedIn {
                    ElectroButton(text: L("Заказать")) { showForm = true }
                        .padding(.horizontal, Space.x5).padding(.bottom, Space.x4)
                } else {
                    Text(L("Войдите в аккаунт, чтобы оставить заявку")).font(ElectroType.caption).foregroundStyle(p.textMuted)
                        .frame(maxWidth: .infinity).padding(Space.x5)
                }
            }
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(p.background)
    }

    private func qtyButton(_ icon: String, _ action: @escaping () -> Void) -> some View {
        Button(action: action) {
            Image(systemName: icon).font(.system(size: 15, weight: .semibold)).foregroundStyle(p.textPrimary)
                .frame(width: 36, height: 36).background(p.surfaceElevated).clipShape(Circle())
        }
        .buttonStyle(.plain)
    }
}

/// Кнопка «Магазин» — сумка в кругляше, рядом с колокольчиком.
struct ShopFab: View {
    @Environment(\.palette) private var p
    let action: () -> Void
    var body: some View {
        Button(action: action) {
            Image(systemName: "bag").font(.system(size: 18, weight: .medium)).foregroundStyle(p.textSecondary)
                .frame(width: 44, height: 44)
                .background(p.surfaceElevated)
                .clipShape(Circle())
        }
        .buttonStyle(.plain)
        .accessibilityLabel(L("Магазин"))
    }
}

File: android/app/src/main/java/uz/electro/remote/ui/ShopScreen.kt
package uz.electro.remote.ui

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.platform.LocalUriHandler
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextDecoration
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import uz.electro.remote.data.ProductDto
import uz.electro.remote.i18n.S
import uz.electro.remote.ui.components.PullRefresh
import uz.electro.remote.ui.components.ButtonStyle
import uz.electro.remote.ui.components.ElectroButton
import uz.electro.remote.ui.components.Lx
import uz.electro.remote.ui.theme.*

/**
 * Магазин аксессуаров и доп. оборудования: витрина карточками, карточка товара с
 * описанием и заявкой (телефон + комментарий). Оплаты в приложении нет — менеджер
 * перезванивает. Без входа витрину видно, заказать — только после входа.
 */
@Composable
fun ShopScreen(
    products: List<ProductDto>,
    loggedIn: Boolean,
    phoneHint: String,
    model: String?,
    onOrder: (productId: String, qty: Int, phone: String, comment: String, done: (String?) -> Unit) -> Unit,
    onBack: () -> Unit,
    onRefresh: (() -> Unit)? = null,
) {
    var category by remember { mutableStateOf("all") }
    var query by remember { mutableStateOf("") }
    var selected by remember { mutableStateOf<ProductDto?>(null) }
    val categories = listOf(
        "all" to S("Все"), "accessory" to S("Аксессуары"), "equipment" to S("Оборудование"),
        "care" to S("Уход"), "tuning" to S("Тюнинг"), "other" to S("Другое"),
    )
    selected?.let { p ->
        androidx.activity.compose.BackHandler { selected = null }
        ProductScreen(p, loggedIn, phoneHint, onOrder, onClose = { selected = null })
        return
    }
    // сначала товары для своей модели, потом общие, потом остальные
    val m = (model ?: "").uppercase()
    val q = query.trim().lowercase()
    val shown = products
        .filter { category == "all" || it.category == category }
        .filter { q.isEmpty() || it.title.lowercase().contains(q) || it.description.lowercase().contains(q) || it.models.lowercase().contains(q) }
        .sortedBy { p -> when { p.models.isBlank() -> 1; m.isNotBlank() && p.models.uppercase().contains(m) -> 0; else -> 2 } }

    PullRefresh(onRefresh) {
    ScreenScaffold(S("Магазин"), onBack) {
        OutlinedTextField(
            value = query, onValueChange = { query = it }, singleLine = true,
            placeholder = { Text(S("Поиск товаров"), color = ElectroColors.TextMuted) },
            leadingIcon = { Icon(Lx.Search, null, tint = ElectroColors.TextMuted, modifier = Modifier.size(20.dp)) },
            trailingIcon = if (query.isNotEmpty()) ({ Icon(Lx.Close, null, tint = ElectroColors.TextMuted,
                modifier = Modifier.size(20.dp).clickable { query = "" }) }) else null,
            shape = Radius.Md, modifier = Modifier.fillMaxWidth(),
            colors = OutlinedTextFieldDefaults.colors(
                focusedBorderColor = ElectroColors.Accent, unfocusedBorderColor = ElectroColors.Outline,
                focusedTextColor = ElectroColors.TextPrimary, unfocusedTextColor = ElectroColors.TextPrimary,
                cursorColor = ElectroColors.Accent,
            ),
        )
        Row(Modifier.horizontalScroll(rememberScrollState()), horizontalArrangement = Arrangement.spacedBy(6.dp)) {
            categories.forEach { (k, label) ->
                val on = category == k
                Surface(
                    color = if (on) ElectroColors.Accent.copy(alpha = 0.14f) else ElectroColors.SurfaceElevated, shape = Radius.Pill,
                    border = if (on) androidx.compose.foundation.BorderStroke(1.dp, ElectroColors.Accent) else null,
                    modifier = Modifier.height(ControlSize.Chip).clip(Radius.Pill).clickable { category = k },
                ) {
                    Box(Modifier.padding(horizontal = Space.x4).fillMaxHeight(), contentAlignment = Alignment.Center) {
                        Text(label, style = ElectroType.Body, color = if (on) ElectroColors.Accent else ElectroColors.TextPrimary, maxLines = 1)
                    }
                }
            }
        }
        if (shown.isEmpty()) EmptyNote(when { products.isEmpty() -> S("Товары скоро появятся."); q.isNotEmpty() -> S("Ничего не найдено"); else -> S("В этом разделе пусто.") })
        // сетка 2 в ряд
        shown.chunked(2).forEach { row ->
            // карточки в ряду одной высоты: строка старой цены зарезервирована всегда
            Row(Modifier.fillMaxWidth().height(IntrinsicSize.Max), horizontalArrangement = Arrangement.spacedBy(Space.x3)) {
                row.forEach { p -> ProductCard(p, Modifier.weight(1f).fillMaxHeight()) { selected = p } }
                if (row.size == 1) Spacer(Modifier.weight(1f))
            }
        }
    }
    }
}

fun formatPrice(price: Long, currency: String): String {
    val n = "%,d".format(price).replace(',', ' ')
    return if (currency == "USD") "$$n" else S("{0} сум", n)
}

@Composable
private fun ProductCard(p: ProductDto, modifier: Modifier, onClick: () -> Unit) {
    Surface(
        color = ElectroColors.Surface, shape = Radius.Md,
        border = androidx.compose.foundation.BorderStroke(1.dp, ElectroColors.Outline),
        modifier = modifier.clip(Radius.Md).clickable(onClick = onClick),
    ) {
        Column(Modifier.padding(Space.x3)) {
            Box(Modifier.fillMaxWidth().aspectRatio(1f).clip(Radius.Sm).background(ElectroColors.SurfaceElevated), contentAlignment = Alignment.Center) {
                if (!p.image_url.isNullOrBlank()) coil.compose.AsyncImage(
                    model = p.image_url, contentDescription = null, contentScale = ContentScale.Crop,
                    modifier = Modifier.fillMaxSize(),
                ) else Icon(Lx.ShoppingBag, null, tint = ElectroColors.TextMuted, modifier = Modifier.size(36.dp))
                if (p.old_price != null && p.old_price > p.price) Box(
                    Modifier.align(Alignment.TopStart).padding(8.dp).clip(Radius.Pill).background(ElectroColors.Accent)
                        .padding(horizontal = 7.dp, vertical = 2.dp),
                ) { Text("-${100 - (p.price * 100 / p.old_price)}%", fontSize = 10.sp, fontWeight = FontWeight.Bold, color = ElectroColors.OnAccent) }
            }
            Spacer(Modifier.height(Space.x2))
            Text(p.title, style = ElectroType.Body, fontWeight = FontWeight.SemiBold, color = ElectroColors.TextPrimary,
                maxLines = 2, minLines = 2, overflow = TextOverflow.Ellipsis)
            Spacer(Modifier.height(4.dp))
            Text(formatPrice(p.price, p.currency), style = ElectroType.Body, fontWeight = FontWeight.Bold, color = ElectroColors.Accent)
            // строка старой цены есть всегда (пустая, если скидки нет) — иначе карточки разной высоты
            val discount = p.old_price != null && p.old_price > p.price
            Text(if (discount) formatPrice(p.old_price!!, p.currency) else " ", style = ElectroType.Unit, color = ElectroColors.TextMuted,
                textDecoration = if (discount) TextDecoration.LineThrough else TextDecoration.None, maxLines = 1)
        }
    }
}

/** Карточка товара: фото, цена, описание, заявка. */
@Composable
private fun ProductScreen(
    p: ProductDto, loggedIn: Boolean, phoneHint: String,
    onOrder: (String, Int, String, String, (String?) -> Unit) -> Unit,
    onClose: () -> Unit,
) {
    val uriHandler = LocalUriHandler.current
    var qty by remember { mutableIntStateOf(1) }
    var phone by remember { mutableStateOf(phoneHint) }
    var comment by remember { mutableStateOf("") }
    var busy by remember { mutableStateOf(false) }
    var sent by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }
    var showForm by remember { mutableStateOf(false) }

    Column(Modifier.fillMaxSize().background(ElectroColors.Background)) {
        Row(Modifier.fillMaxWidth().padding(Space.x4), verticalAlignment = Alignment.CenterVertically) {
            Icon(Lx.ArrowBack, S("Назад"), tint = ElectroColors.TextPrimary, modifier = Modifier.size(26.dp).clickable(onClick = onClose))
            Spacer(Modifier.width(Space.x3))
            Text(S("Товар"), style = ElectroType.Headline, color = ElectroColors.TextPrimary, modifier = Modifier.weight(1f))
        }
        Column(
            Modifier.weight(1f).fillMaxWidth().verticalScroll(rememberScrollState())
                .padding(horizontal = Space.x5).padding(bottom = Space.x6),
            verticalArrangement = Arrangement.spacedBy(Space.x4),
        ) {
            if (!p.image_url.isNullOrBlank()) coil.compose.AsyncImage(
                model = p.image_url, contentDescription = null, contentScale = ContentScale.Crop,
                modifier = Modifier.fillMaxWidth().height(240.dp).clip(Radius.Md).background(ElectroColors.SurfaceElevated),
            )
            Text(p.title, style = ElectroType.Title, color = ElectroColors.TextPrimary)
            Row(verticalAlignment = Alignment.Bottom, horizontalArrangement = Arrangement.spacedBy(Space.x2)) {
                Text(formatPrice(p.price, p.currency), style = ElectroType.Headline, color = ElectroColors.Accent)
                if (p.old_price != null && p.old_price > p.price)
                    Text(formatPrice(p.old_price, p.currency), style = ElectroType.Caption, color = ElectroColors.TextMuted, textDecoration = TextDecoration.LineThrough)
            }
            if (p.models.isNotBlank()) Text(S("Подходит: {0}", p.models), style = ElectroType.Caption, color = ElectroColors.TextSecondary)
            if (p.description.isNotBlank()) Text(p.description, style = ElectroType.Body, color = ElectroColors.TextSecondary)
            if (!p.link.isNullOrBlank()) Text(S("Подробнее на сайте"), style = ElectroType.Body, color = ElectroColors.Accent,
                fontWeight = FontWeight.SemiBold, modifier = Modifier.clickable { runCatching { uriHandler.openUri(p.link) } })

            if (sent) {
                uz.electro.remote.ui.components.ElectroToast(uz.electro.remote.ui.components.BadgeKind.Success,
                    S("Заявка отправлена"), S("Менеджер свяжется с вами по указанному телефону."))
            } else if (showForm) {
                Surface(color = ElectroColors.Surface, shape = Radius.Md, border = androidx.compose.foundation.BorderStroke(1.dp, ElectroColors.Outline)) {
                    Column(Modifier.padding(Space.x4), verticalArrangement = Arrangement.spacedBy(Space.x3)) {
                        Text(S("Заявка на покупку"), style = ElectroType.Body, fontWeight = FontWeight.SemiBold, color = ElectroColors.TextPrimary)
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            Text(S("Количество"), style = ElectroType.Body, color = ElectroColors.TextSecondary, modifier = Modifier.weight(1f))
                            QtyButton("−") { if (qty > 1) qty-- }
                            Text("$qty", style = ElectroType.Headline, color = ElectroColors.TextPrimary, modifier = Modifier.padding(horizontal = Space.x3))
                            QtyButton("+") { if (qty < 99) qty++ }
                        }
                        Field(S("ТЕЛЕФОН"), phone, { phone = it }, KeyboardType.Phone)
                        Text(S("КОММЕНТАРИЙ (необязательно)"), color = ElectroColors.TextSecondary, fontSize = 12.sp, fontWeight = FontWeight.SemiBold)
                        OutlinedTextField(value = comment, onValueChange = { comment = it }, minLines = 2, modifier = Modifier.fillMaxWidth(),
                            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Text))
                        error?.let { Text(it, color = ElectroColors.Danger, style = ElectroType.Caption) }
                        ElectroButton(if (busy) S("Отправляем…") else S("Отправить заявку"), Modifier.fillMaxWidth(), enabled = !busy && phone.trim().length >= 5) {
                            busy = true; error = null
                            onOrder(p.id, qty, phone.trim(), comment.trim()) { err -> busy = false; if (err == null) sent = true else error = err }
                        }
                    }
                }
            }
        }
        if (!sent && !showForm) {
            if (loggedIn) ElectroButton(S("Заказать"), Modifier.fillMaxWidth().padding(horizontal = Space.x5).padding(bottom = Space.x4)) { showForm = true }
            else Text(S("Войдите в аккаунт, чтобы оставить заявку"), style = ElectroType.Caption, color = ElectroColors.TextMuted,
                modifier = Modifier.fillMaxWidth().padding(Space.x5), textAlign = androidx.compose.ui.text.style.TextAlign.Center)
        }
    }
}

@Composable
private fun QtyButton(label: String, onClick: () -> Unit) {
    Surface(onClick = onClick, shape = CircleShape, color = ElectroColors.SurfaceElevated, modifier = Modifier.size(36.dp)) {
        Box(Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
            Text(label, fontSize = 20.sp, color = ElectroColors.TextPrimary)
        }
    }
}

/** Кнопка «Магазин» — сумка в кругляше, рядом с колокольчиком. */
@Composable
fun ShopFab(onClick: () -> Unit, modifier: Modifier = Modifier) {
    Surface(onClick = onClick, shape = CircleShape, color = ElectroColors.SurfaceElevated, modifier = modifier.size(44.dp)) {
        Box(Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
            Icon(Lx.ShoppingBag, S("Магазин"), tint = ElectroColors.TextSecondary, modifier = Modifier.size(22.dp))
        }
    }
}
